app/api/tests/test_api.py

```diff
@@ -1199,6 +1201,119 @@ def test_can_filter_doc_without_anotation(self):
+                                                   password=cls.super_user_pass,
+                                                   email='fizz@buzz.com')
+        cls.classification_project = mommy.make('TextClassificationProject',
+                                                users=[super_user], project_type=DOCUMENT_CLASSIFICATION)
+
+        cls.classification_upload_url = reverse(viewname='doc_uploader', args=[cls.classification_project.id])
+        cls.classification_download_url = reverse(viewname='doc_downloader', args=[cls.classification_project.id])
+        assign_user_to_role(project_member=super_user, project=cls.classification_project,
+                            role_name=settings.ROLE_PROJECT_ADMIN)
+
+    def setUp(self):
+        self.client.login(username=self.super_user_name,
+                          password=self.super_user_pass)
+
+    def load_test_helper(self, upload_url, download_url, filename, import_format, project, export_format, response_format="text/csv; charset=utf-8",**kwargs):
+        import_data = []
+        parser = TextUploadAPI.select_parser(import_format)
+        with open(os.path.join(DATA_DIR, filename), 'rb') as f:
+            self.client.post(upload_url, data={'file': f, 'format': import_format})
+            f.seek(0)
+            imported = parser.parse(f)
+            import_data = [elem for elem in [x for elem in imported for x in elem] if elem.get('labels')]
+
+        r = self.client.get(download_url, data={'q': export_format}, HTTP_ACCEPT=response_format)
+        b = BytesIO(r.content)
+        if export_format == 'txt':
+            export_format = import_format
+        parser = TextUploadAPI.select_parser(export_format)
+        exported = parser.parse(b)
+        exported_data = [x for x in [x for elem in exported for x in elem] if x.get('labels') or x.get('annotations')]
+        self.assertTrue(len(import_data) == len(exported_data), 'Length of imported dataset does not match exported')
+        return import_data, exported_data
+
+    # Classification
+    def test_jsonl_classification_import_export_integrity(self):
+        import_data, export_data = self.load_test_helper(upload_url=self.classification_upload_url,
+                                download_url=self.classification_download_url,
+                                filename='classification.jsonl',
+                                import_format='json',
+                                project=self.classification_project,
+                                export_format='json',
+                                response_format='application/json')
+        label_mapping = {label.id: label.text for label in self.classification_project.labels.all()}
+        for im, ex in zip(import_data, export_data):
+            self.assertTrue(im['text'] == ex['text'], 'Integritycheck failed. Items differ.')
+            ex_labels = set([label_mapping[int(x)] for x in [x.get('label') for x in ex.get('annotations')]])
```

Comment:
simplify: `set(label_mapping[int(x.get('label'))] for x in ex.get('annotations', []))`